pandas/core/groupby/ops.py

```diff
@@ -212,9 +212,12 @@ def apply(self, f, data, axis=0):
                 # This Exception is also raised if `f` triggers an exception
                 # but it is preferable to raise the exception in Python.
                 pass
-            except Exception:
-                # raise this error to the caller
-                pass
+            except TypeError as err:
+                if "Cannot convert" in str(err):
+                    # via apply_frame_axis0 if we pass a non-ndarray
+                    pass
```

Comment:
looks fine, does this change any perf? IOW this looks like this is now taking a path that previously we raised (and then likely did an .apply on)

Reply:
Not sure.  Between this, #27909, and a not-yet-pushed branch that fixes incorrect exception handling in cython_agg_block, I'm pretty sure we'll end up falling back to python-space less often, but it isn't obvious what the individual changes affect perf-wise.